Bottom: 65ec346f8ca209df364353d8fb261901fe6c442c
Top:    cbe09fc5f1bd22481c10d400d68ecd0225165777
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-07 00:56:48 +0000

Always initialise default layernames for the PCB (silk, soldermask etc..)

This avoids retaining the default PCB stackup layer names when loading a
board which has missing layers. (Such as silkscreen or soldermask).


---

diff --git a/src/file.c b/src/file.c
index 11ab7c3..3abca69 100644
--- a/src/file.c
+++ b/src/file.c
@@ -408,7 +408,7 @@ real_load_pcb (char *Filename, bool revert)
     {
       RemovePCB (oldPCB);
 
-      CreateNewPCBPost (PCB, 0);
+      CreateNewPCBPost (PCB, 1);
       ResetStackAndVisibility ();
 
       /* update cursor location */
diff --git a/src/parse_y.y b/src/parse_y.y
index 629ccd6..fe675a2 100644
--- a/src/parse_y.y
+++ b/src/parse_y.y
@@ -196,7 +196,7 @@ parsepcb
 			{
 			  PCBType *pcb_save = PCB;
 
-			  CreateNewPCBPost (yyPCB, 0);
+			  CreateNewPCBPost (yyPCB, 1);
 			/* initialize the polygon clipping now since
 			 * we didn't know the layer grouping before.
 			 */
@@ -226,7 +226,7 @@ parsepcb
 		    if (yyPCB != NULL)
 		      {
 			/* This case is when we load a footprint with file->open, or from the command line */
-			CreateNewPCBPost (yyPCB, 0);
+			CreateNewPCBPost (yyPCB, 1);
 			ParseGroupString("1,c:2,s", &yyPCB->LayerGroups, &yyData->LayerN);
 			e = yyPCB->Data->Element->data; /* we know there's only one */
 			PCB = yyPCB;
